### PR TITLE
Fork diag/code-action tests to ones without editor dep. 1/N

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Depends on https://github.com/dotnet/roslyn/pull/71551

The general plan is:

Fork existing test helpers at editor layer, and copy them down to features layer.  In that step, remove any editor dependencies (things like 'copy/paste', or 'preview window', etc.).  That is this PR.

Move tests that reference the editor helpers unnecessarily, down to the features layer.  That is this PR.  For C# code refactoring tests.

In the future:

- Break up more test helpers.  Like the ones for diagnostics + analyzers.
- Move more tests down.
- Take the helpers that depend on the editor and have them use the new helpers (like with inheritance).

### CODE REVIEW

File: src/Features/DiagnosticsTestUtilities/CodeActionsLegacy/AbstractCodeActionOrUserDiagnosticTest_NoEditor_OptionHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions
+{
+    public abstract partial class AbstractCodeActionOrUserDiagnosticTest_NoEditor
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/CodeActionsLegacy/AbstractCodeActionOrUserDiagnosticTest_NoEditor_OptionHelpers.cs:11]: these are copies for now, with the editor parts stripped out.  A followup PR will take the tyeps that still use the editor an dhave them layer back onto this.

File: src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
   <ItemGroup Label="Project References">
     <!-- TODO: Remove the below project references to test utility projects once all analyzer/code fix tests are switched to Microsoft.CodeAnalysis.Testing -->
     <ProjectReference Include="..\..\..\EditorFeatures\TestUtilities\Microsoft.CodeAnalysis.EditorFeatures.Test.Utilities.csproj" />
+    <ProjectReference Include="..\..\..\Features\TestUtilities\Microsoft.CodeAnalysis.Features.Test.Utilities.csproj"/>
```
Comment (CyrusNajmabadi) [inline src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj:30]: new projects at the Features layer have been added to mirror the structure at the EditorFeatures layer.  Types in EditorFeatures that don't need to be there are moving down.

**Comment by CyrusNajmabadi** [comment]
@tmat @genlu ptal.

File: src/Features/TestUtilities/Microsoft.CodeAnalysis.Features.Test.Utilities.csproj [hunk 3]
```diff
@@ -1,10 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
-<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
```
Comment (genlu) [inline src/Features/TestUtilities/Microsoft.CodeAnalysis.Features.Test.Utilities.csproj:3]: Why this change?

Comment (CyrusNajmabadi) [reply src/Features/TestUtilities/Microsoft.CodeAnalysis.Features.Test.Utilities.csproj:3]: This project was actually not added to the solution. So this is effectively a full file add. I brought this up to date with the equivalent EditorFeatures version.  Good possibly change that value though if not desired.

**Review by genlu — APPROVED** [review]
